puma/service/templates/puma.rb.j2

```diff
@@ -23,8 +23,21 @@ on_restart do
   ENV["BUNDLE_GEMFILE"] = "{{ puma_gemfile }}"
 end
 
+before_fork do
+  if defined?(::ActiveRecord) && defined?(::ActiveRecord::Base)
+    ActiveRecord::Base.connection_pool.disconnect!
+  end
+end
+
 on_worker_boot do
-  ActiveSupport.on_load(:active_record) do
+  if defined?(::ActiveRecord) && defined?(::ActiveRecord::Base)
     ActiveRecord::Base.establish_connection
   end
+  if defined?(CustomPumaConfig) && CustomPumaConfig.respond_to?(:on_worker_boot)
+    begin
+      CustomPumaConfig.on_worker_boot
+    rescue Exception
+      STDERR.puts "Error while loading custom puma config\n#{$!.message}\n\t#{$!.backtrace.join("\n\t")}".gsub(/^$\n/, '')
```

Comment:
As far as i know, yes. We need to make absolutly sure, that this won't hinder the start up process of a worker.